service/common/src/main/java/org/apache/polaris/service/admin/PolarisServiceImpl.java

```diff
@@ -142,12 +211,24 @@ private PolarisAdminService newAdminService(
   @Override
   public Response createCatalog(
       CreateCatalogRequest request, RealmContext realmContext, SecurityContext securityContext) {
+    String requestId = PolarisEvent.createRequestId();
```

Comment:
A few remarks:

Injecting before/after event logic in every method looks like we are mixing concerns. I think the decorator pattern could make things look a lot nicer.

About the execution flow: how do we handle errors?

E.g. if the listener throws on a before event call, should the execution halt or continue?

Also: if the listener throws on an after event call, should the execution be aborted and the persisted changes rolled back, or should the business changes be kept? IOW, are we OK with weaker delivery guarantees for events (like "at most once")?